core/src/main/scala/spinal/core/Vec.scala

```diff
@@ -279,6 +279,14 @@ class Vec[T <: Data](val dataType: HardType[T], val vec: Vector[T]) extends Mult
     elementsCache
   }
 
+  def shuffle(indexMapping: (Int, Int) => Int): Vec[T] = {
```

Comment:
Why the `indexMapping` function takes two integer params?

Reply:
one for total length, one for current index.

Reply:
I think the shuffle in tester would explain more.

Reply:
The name of the `indexMapping` function implies mapping one index to another index. Two integer params may confuse users who doesn't chain `shuffle` after `subdividenIn`, I think. 
Say your test case, `total` can be `data.getWidth/8` and not a necessary param.

Reply:
An alternative implemenation of your test example, 
```scala
  outData := data.subdivideIn(8 bits).shuffle { i =>
    val total = data.getWidth / 8
    total - 1 -i
  }
```

Reply:
i agree that the second intparameter seems unatural. (i understand it can be usefull, but then i would say it may be proposed into a second cuntion, ex shuffleWithSize)